org.eclipse.scout.rt.jackson/src/main/java/org/eclipse/scout/rt/jackson/dataobject/DoEntitySerializer.java

```diff
@@ -121,10 +121,12 @@ else if (obj.getClass() == DoEntity.class) {
     }
     else {
       Optional<AttributeType> attributeType = getAttributeType(attributeName);
-      if (attributeType.isPresent()) {
+      if (attributeType.isPresent() && (!m_context.isLenientMode() || attributeType.get().getJavaType().isTypeOrSuperTypeOf(obj.getClass()))) {
+        // serialize by typed attribute if a type is present and if lenient, only if type matches (data object might have an invalid structure, e.g. string instead of an enum if deserialized lenient)
         serializeTypedAttribute(attributeName, obj, gen, provider, attributeType.get().getJavaType());
       }
       else {
+        // serialize by instance
```

Comment:
"value-based serialization" or "serialize by instance value"?  (e.g. value-based vs. class-based serialization)

Reply:
Changed to "use serialization by value" as used in other suggestions